gnucash/import-export/import-main-matcher.c

```diff
@@ -147,36 +147,27 @@ static void refresh_model_row (
 
 static gboolean delete_hash (gpointer key, gpointer value, gpointer user_data)
 {
-    // Value is a hash table that needs to be destroyed. JEAN: I should free the memory used for the keys.
+    // Value is a hash table that needs to be destroyed. 
     g_hash_table_destroy (value);
     return TRUE;
 }
 
 static void
-commit_all (GNCImportMainMatcher *info)
+update_all_balances (GNCImportMainMatcher *info)
 {
-    // Commit the edits for all accounts for which we've called inc_account_edit.
     for (GSList* iter = info->edited_accounts; iter; iter=iter->next)
-    {
-        int level = qof_instance_get_editlevel (iter->data);
-        // This should not happen.
-        if (level != 1)
-            PERR ("import-main-macher.c: Edit level not 1");
-        while (qof_instance_get_editlevel (iter->data) > 0)
-            xaccAccountCommitEdit (iter->data);
-    }
+        gnc_account_set_defer_bal_computation(iter->data,FALSE);
     g_slist_free (info->edited_accounts);
     info->edited_accounts = NULL;
 }
 
 static void
-inc_account_edit (GNCImportMainMatcher *info, Account* acc)
+defer_bal_computation (GNCImportMainMatcher *info, Account* acc)
 {
-    // Call BeginEdit on accounts that haven't been edited yet. This will speeds things up
-    // a lot by preventing commits until we're all done.
-    if (qof_instance_get_editlevel (acc) == 0)
+    if (!gnc_account_get_defer_bal_computation(acc))
     {
-        xaccAccountBeginEdit (acc);
+        gnc_account_set_defer_bal_computation (acc, TRUE);
+        xaccAccountRecomputeBalance (acc);
```

Comment:
Doesn't this belong in `update all balances`?

Reply:
Well, first of `xaccAccountRecomputeBalance (acc);` should not be there, for one thing it does nothing, and second it's irrelevant here, since we're trying to avoid any balance recomputation!
But `defer_bal_computation` is called from multiple places to set the flag on the accounts, and add them to our list so we can reset the flags at the end.
Or maybe I don't understand your comment.

Reply:
I think you just put the `xaccAccountRecomputeBalance (acc)` call in the wrong place. Doesn't it belong in `update_all_balances` after the call to `gnc_account_set_defer_bal_computation (acc, FALSE)` call?

Reply:
You're absolutely right. Dang it, I need to be more careful. Thanks.